packages/dds/tree/src/util/nestedMap.ts

```diff
@@ -41,17 +41,30 @@ export function tryAddToNestedMap<Key1, Key2, Value>(
 
 /**
  * Copies over all entries from the source map into the destination map.
+ * By default, does not override any existing values.
  *
  * @internal
  */
 export function populateNestedMap<Key1, Key2, Value>(
 	source: NestedMap<Key1, Key2, Value>,
 	destination: NestedMap<Key1, Key2, Value>,
+	override = false,
 ): void {
 	for (const [key1, innerMap] of source) {
-		destination.set(key1, new Map(innerMap));
+		const newInner = new Map(destination.get(key1));
+
+		for (const [key2, value] of innerMap) {
+			if (override || !newInner.has(key2)) {
+				newInner.set(key2, value);
+			}
+		}
+
+		if (newInner.size > 0) {
+			destination.set(key1, newInner);
+		}
```

Comment:
Do we really need this?
`newInner.size === 0` should only occur if source or destination have empty inner maps.

Note that with this check...
* destination might still end up with missing nested maps if it started with them.
* destination might start with a missing nested map but have that nested map removed.
The combination of these two points makes for a confusing contract.

It seems questionable for this function to legislate on whether or not empty inner maps are acceptable.

Reply:
Oh that's a good point, I was just worried about adding an empty map when there wasn't already one but yeah, that shouldn't be possible